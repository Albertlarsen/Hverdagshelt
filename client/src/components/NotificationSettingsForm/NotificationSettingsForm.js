// @flow

import React from 'react';
import {CategoryService, NotificationSettingsService} from "../../services";
import jwt from "jsonwebtoken";
import FormGroup from "react-bootstrap/es/FormGroup";
import FormControl from "react-bootstrap/es/FormControl";
import ControlLabel from "react-bootstrap/es/ControlLabel";
import Checkbox from "react-bootstrap/es/Checkbox";
import Button from "react-bootstrap/es/Button";

let notificationSettingsService = new NotificationSettingsService();
let categoryService = new CategoryService();


interface State {
    decoded: Object,
    notificationCategories: Object[],
    chosenNotifications: Object[]
}

export class NotificationSettingsForm extends React.Component <State> {
    state = {
        decoded: jwt.verify(window.localStorage.getItem('userToken'), "shhhhhverysecret"),
        notificationCategories: [],
        chosenNotifications: []
    };


    handleChange(e) {
        this.setState({value: e.target.value});
    }

    componentDidMount() {
        let chosenCat = [];
        let categories = [];

        notificationSettingsService.getNotificationSettings(this.state.decoded.email)
            .then(resources => {
                resources.map(e => (
                    chosenCat.push(e.name)
                ));
                this.setState({
                    chosenNotifications: chosenCat
                });
            });

        categoryService.getCategory1()
            .then(resources => {
                resources.map(e => (
                    categories.push(e.name)
                ));
                this.setState({
                    notificationCategories: categories
                });
            });

    }

    render() {
        return (
            <form>
                <h3>Velg kategorier du vil har varselmeldinger p√•:</h3>
                {
                    this.state.notificationCategories.map(e =>
                        this.state.chosenNotifications.includes(e) ? (
                            <div key={e}>
                                <FormGroup id={e}>
                                    <Checkbox checked>{e}</Checkbox>
                                </FormGroup>
                            </div>
                        ) : (
                            <div key={e}>
                                <FormGroup id={e}>
                                    <Checkbox>{e}</Checkbox>
                                </FormGroup>
                            </div>
                        )
                    )
                }
                <Button>Lagre</Button>
            </form>
        );
    }
}