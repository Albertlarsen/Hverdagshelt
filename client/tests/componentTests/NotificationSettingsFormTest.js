// @flow

import {shallow} from "enzyme/build";
import * as React from "react";
import {NotificationSettingsForm} from "../../src/components/NotificationSettingsForm/NotificationSettingsForm";

describe('Test for NotificationSettingsForm component', () => {
    const wrapper = shallow(<NotificationSettingsForm />);


    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

});