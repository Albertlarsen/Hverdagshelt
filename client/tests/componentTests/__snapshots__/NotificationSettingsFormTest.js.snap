// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for NotificationSettingsForm component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotificationSettingsForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Grid
        bsClass="container"
        componentClass="div"
        fluid={false}
      >
        <h3>
          Velg kategorier du vil har varselmeldinger på:
        </h3>
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="primary"
          disabled={false}
          onClick={[Function]}
        >
          Lagre
        </Button>
      </Grid>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "container",
        "children": Array [
          <h3>
            Velg kategorier du vil har varselmeldinger på:
          </h3>,
          Array [],
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="primary"
            disabled={false}
            onClick={[Function]}
          >
            Lagre
          </Button>,
        ],
        "componentClass": "div",
        "fluid": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Velg kategorier du vil har varselmeldinger på:",
          },
          "ref": null,
          "rendered": "Velg kategorier du vil har varselmeldinger på:",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "block": false,
            "bsClass": "btn",
            "bsStyle": "primary",
            "children": "Lagre",
            "disabled": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Lagre",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Grid
          bsClass="container"
          componentClass="div"
          fluid={false}
        >
          <h3>
            Velg kategorier du vil har varselmeldinger på:
          </h3>
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="primary"
            disabled={false}
            onClick={[Function]}
          >
            Lagre
          </Button>
        </Grid>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "container",
          "children": Array [
            <h3>
              Velg kategorier du vil har varselmeldinger på:
            </h3>,
            Array [],
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsStyle="primary"
              disabled={false}
              onClick={[Function]}
            >
              Lagre
            </Button>,
          ],
          "componentClass": "div",
          "fluid": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Velg kategorier du vil har varselmeldinger på:",
            },
            "ref": null,
            "rendered": "Velg kategorier du vil har varselmeldinger på:",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "active": false,
              "block": false,
              "bsClass": "btn",
              "bsStyle": "primary",
              "children": "Lagre",
              "disabled": false,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Lagre",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
